src/binder/bind/bind_projection_clause.cpp

```diff
@@ -131,6 +131,27 @@ std::pair<expression_vector, std::vector<std::string>> Binder::bindProjectionLis
     return {projectionExprs, aliases};
 }
 
+static void validateNestedAggregate(const Expression& expr, const BinderScope& scope) {
+    KU_ASSERT(expr.expressionType == ExpressionType::AGGREGATE_FUNCTION);
+    if (expr.getNumChildren() == 0) { // Skip COUNT(*)
+        return;
+    }
+    auto collector = AggregateExprCollector();
+    collector.visit(expr.getChild(0));
+    auto aggs = collector.getAggregates();
+    auto numAggs = 0u;
+    for (auto& agg : aggs) {
+        if (scope.contains(agg->getAlias())) {
+            continue;
+        }
+        numAggs++;
+    }
+    if (numAggs > 0) {
+        throw BinderException(
+            stringFormat("Expression {} contains nested aggregation.", expr.toString()));
+    }
```

Comment:
```suggestion
    for (auto& agg : aggs) {
        if (!scope.contains(agg->getAlias())) {
            throw BinderException(
                stringFormat("Expression {} contains nested aggregation.", expr.toString()));
        }
    }
```

Seems L142- L152 can be simplified.